File: src/main/kotlin/org/kravbank/java/model/RequirementVariant.java
package org.kravbank.java.model;

import io.quarkus.hibernate.orm.panache.PanacheEntity;
import javax.persistence.Entity;
import javax.persistence.OneToMany;
import java.util.List;

@Entity
public class RequirementVariant extends PanacheEntity {
    public String description;

    public String requirementText;

    public String instruction;

    public Boolean useProduct;

    public Boolean useSpesification; //typo

    public Boolean useQualification;

    @OneToMany
    public List<Project> products;

    @OneToMany
    public List <Config> configs;


    //public questions
    // public String type;


    //CONFIG


}

File: src/main/kotlin/org/kravbank/domain/RequirementVariant.kt
package org.kravbank.domain

import com.fasterxml.jackson.annotation.JsonBackReference
import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonManagedReference
import io.quarkus.hibernate.orm.panache.PanacheEntity
import java.util.*
import javax.persistence.*

@Entity
@JsonIgnoreProperties(ignoreUnknown = true)
class RequirementVariant : PanacheEntity() {

    @Column(unique = true)
    var ref: String = UUID.randomUUID().toString()

    lateinit var description: String

    lateinit var requirementText: String

    lateinit var instruction: String

    var useProduct: Boolean = false

    var useSpecification: Boolean = false

    var useQualification: Boolean = false


    @ManyToOne(
        cascade = [CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REFRESH], fetch = FetchType.LAZY
    )
    @JsonManagedReference(value = "val-requirementVariant")
    @JsonIgnore
    var requirement: Requirement? = null

    @ManyToMany(
        cascade = [CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH],
    )
    @JsonBackReference(value = "val-reqvariant-product")
    var products: MutableList<Product> = mutableListOf()

}